microcosm_flask/swagger/parameters/base.py

```diff
@@ -0,0 +1,142 @@
+            "items": self.parse_items,
+        }
+
+    @classmethod
+    def build(cls, field: Field) -> Mapping:
+        """
+        Build a swagger parameter from a marshmallow field.
+
+        """
+        builders = [
+            entry_point.load()()
+            for entry_point in iter_entry_points(ENTRY_POINT)
+        ] + [
+            # put default last
+            cls.default_builder_type()(),
+        ]
+
+        builder = next(
+            builder
+            for builder in builders
+            if builder.supports_field(field)
+        )
+
+        return builder.build_parameter(field)
+
+    @classmethod
+    def default_builder_type(cls):
+        """
+        Define the default builder type.
+
+        """
+        # avoid circular dependencies
+        from microcosm_flask.swagger.parameters.default import DefaultParameterBuilder
+        return DefaultParameterBuilder
+
+    def build_parameter(self, field: Field) -> Mapping:
+        """
+        Build a parameter.
+
+        """
+        return dict(self.iter_parsed_values(field))
+
+    def iter_parsed_values(self, field: Field) -> Iterable[Tuple[str, Any]]:
+        """
+        Walk the dictionary of parsers and emits all non-null values.
+
+        """
+        for key, func in self.parsers.items():
+            value = func(field)
+            if not value:
+                continue
+            yield key, value
+
+    @abstractmethod
+    def supports_field(self, field: Field) -> bool:
+        """
+        Does this builder support this kind of field?
+
+        """
+        return False
+
+    @abstractmethod
+    def parse_type(self, field: Field) -> Optional[str]:
+        return None
+
+    def parse_default(self, field: Field) -> Any:
+        """
+        Parse the default value for the field, if any.
+
+        """
+        return field.default
+
+    def parse_description(self, field: Field) -> Optional[str]:
+        """
+        Parse the description for the field, if any.
+
+        """
+        return field.metadata.get("description")
+
+    def parse_enum_values(self, field: Field) -> Optional[Sequence]:
+        """
+        Parse enumerated value for enum fields, if any.
+
+        """
+        return None
+
+    def parse_format(self, field: Field) -> Optional[str]:
+        """
+        Parse the format for the field, if any.
+
+        """
+        return None
+
+    def parse_items(self, field: Field) -> Optional[Mapping]:
+        """
+        Parse the child item type for list fields, if any.
+
+        """
+        if not isinstance(field, List):
+            return None
+
+        return self.build(field.container)
+
+    def parse_ref(self, field: Field) -> Optional[str]:
+        """
+        Parse the reference type for nested fields, if any.
+
+        """
+        if not isinstance(field, Nested):
+            return None
+
+        ref_name = type_name(name_for(field.schema))
+        return f"#/definitions/{ref_name}"
```

Comment:
I'd be tempted to put this and the above into separate builders

Reply:
Good idea.